peerdid/peer_did_utils.py

```diff
@@ -0,0 +1,221 @@
+    """
+    Adds prefix to a data
+    :param key_type: type of key
+    :param data: data to be prefixed
+    :return: prefixed data
+    """
+    prefix = varint.encode(key_type.value)
+    return b''.join([prefix, data])
+
+
+def _encode_filename(filename: str) -> str:
+    """
+    Encodes filename to SHA256 string
+    :param filename: name of file
+    :return: encoded filename as SHA256 string
+    """
+    return hashlib.sha256(filename.encode()).hexdigest()
+
+
+def _build_did_doc_numalgo_0(peer_did: PEER_DID) -> dict:
+    """
+    Helper method to create did_doc according to numalgo 0
+    :param peer_did: peer_did to resolve
+    :raises ValueError:
+    1) if peer_did contains encryption key instead of signing
+    2) if peer_did contains unsupported transform part
+    :return: did_doc
+    """
+    inception_key = peer_did[11:]
+    transform = peer_did[10]
+    if not transform == 'z':
+        raise ValueError(f'Unsupported transform part of peer_did: {transform}')
+    decoded_encnumbasis = _decode_encnumbasis(inception_key, peer_did)
+    if not decoded_encnumbasis['type'] in PublicKeyTypeAuthentication.__members__:
+        raise ValueError('Invalid key type (encryption instead of signing)')
+    did_doc = {
+        'id': peer_did,
+        'authentication': decoded_encnumbasis
+    }
+    return did_doc
+
+
+def _build_did_doc_numalgo_2(peer_did: PEER_DID) -> dict:
+    """
+    Helper method to create did_doc according to numalgo 2
+    :param peer_did: peer_did to resolve
+    :return: did_doc
+    """
+    keys = peer_did[11:]
+    keys = keys.split('.')
+    services = ''
+    keys_without_purpose_code = []
+    for key in keys:
+        if key[0] != 'S':
+            keys_without_purpose_code.append(key[2:])
+        else:
+            services = key[1:]
+    decoded_encnumbasises = [_decode_encnumbasis(key, peer_did) for key in keys_without_purpose_code]
+    decoded_services = _decode_service(services, peer_did)
+
+    authentication = []
+    key_agreement = []
+    for i in range(len(decoded_encnumbasises)):
+        if decoded_encnumbasises[i]['type'] in PublicKeyTypeAuthentication.__members__ and keys[i][0] == 'V':
+            authentication.append(decoded_encnumbasises[i])
+        elif decoded_encnumbasises[i]['type'] in PublicKeyTypeAgreement.__members__ and keys[i][0] == 'E':
+            key_agreement.append(decoded_encnumbasises[i])
+        else:
+            raise ValueError(f'Invalid key type of: {keys[i]}')
+    did_doc = {
+        'id': peer_did,
+        'authentication': authentication,
+        'keyAgreement': key_agreement,
+        'service': decoded_services
+    }
+    return did_doc
+
+
+def _check_key_encoding(key: str, encoding_type: EncodingType) -> bool:
+    """
+    Checks if key correctly encoded
+    :param key: any string
+    :param encoding_type: encoding type
+    :return: true if key correctly encoded, otherwise false
```

Comment:
It never returns False. It throws exceptions instead. Let's remove the exceptions and return False, as the create_peer_did methods expect.